awkward-cpp/include/awkward/io/json.h

```diff
@@ -135,20 +135,91 @@ namespace awkward {
     inline int64_t find_key(const char* str) noexcept {
       int64_t* offsets = string_offsets_.data();
       char* chars = characters_.data();
+      int64_t i;
+      int64_t j;
       int64_t stringi;
       int64_t start;
       int64_t stop;
-      for (int64_t i = current_instruction_ + 1;  i <= current_instruction_ + argument1();  i++) {
+      uint64_t chunkmask;
+      // optimistic: fields in data are in the order specified by the schema
+      if (argument1() != 0) {
+        // increment the current (last seen) field with wrap-around
+        record_current_field_[argument2()]++;
+        if (record_current_field_[argument2()] == argument1()) {
+          record_current_field_[argument2()] = 0;
+        }
+        j = record_current_field_[argument2()];
+        // use the record_current_field_ (as j)
+        i = current_instruction_ + 1 + j;
         stringi = instructions_.data()[i * 4 + 1];
         start = offsets[stringi];
         stop = offsets[stringi + 1];
         if (strncmp(str, &chars[start], (size_t)(stop - start)) == 0) {
-          return instructions_.data()[i * 4 + 2];
+          // ensure that the checklist bit is 1
+          chunkmask = (uint64_t)1 << (j & 0x3f);
```

Comment:
What's the intention of the mask with `0x3f`? It will ensure that `j & 0x3f` is always (0 <= j < 64), but one could also impose `argument2() < 64`. I don't know enough about this logic beyond asking this question :)

Reply:
It's because you could have more than 64 fields in a RecordArray, and `& 0x3f` is `% 64`. I'd have to remind myself of the logic, but this is part of dealing with batches.

Reply:
Right, my question was more concerning what should happen when you get `j > 64`, i.e. why wrap around? In any case, if it passes the tests, that's good enough for me.

Reply:
Ummm. (It's been a while since I wrote it, and I can't easily look at the code now.) There was a reason to have both a division by 64 and a modulo by 64; the bit-shift is the division and the bit-mask is the modulo.

The tests do test this, since they include records with more than 64 fields.